pkg/pgmodel/querier/row.go

```diff
@@ -1,148 +1,40 @@
 package querier
 
 import (
-	"encoding/binary"
 	"sync"
 
-	"github.com/jackc/pgtype"
+	"github.com/jackc/pgx/v5/pgtype"
 	"github.com/prometheus/prometheus/model/labels"
 	"github.com/timescale/promscale/pkg/log"
 	"github.com/timescale/promscale/pkg/pgmodel/common/schema"
 	"github.com/timescale/promscale/pkg/pgmodel/model"
 	"github.com/timescale/promscale/pkg/pgxconn"
 )
 
+// QUESTION(adn) should we cache only the underlying array like it was before?
 var fPool = sync.Pool{
 	New: func() interface{} {
-		return new(pgtype.Float8Array)
+		return new(model.ReusableArray[pgtype.Float8])
 	},
 }
 
 var tPool = sync.Pool{
 	New: func() interface{} {
-		return new(pgtype.TimestamptzArray)
+		return new(model.ReusableArray[pgtype.Timestamptz])
 	},
 }
 
-// wrapper to allow DecodeBinary to reuse the existing array so that a pool is effective
-type timestamptzArrayWrapper struct {
```

Comment:
I moved this to the `custom_types.go` file https://github.com/timescale/promscale/pull/1809/files#diff-d774183a1d65a22e1378babeec3d1e7aed802ebf5f33ed2c5f018d4a3af9a00bR203